src/Microsoft.HttpRepl/Commands/BaseHttpCommand.cs

```diff
@@ -224,10 +233,10 @@ protected override async Task ExecuteAsync(IShellState shellState, HttpState pro
             string bodyTarget = commandInput.Options[ResponseBodyFileOption].FirstOrDefault()?.Text ?? commandInput.Options[ResponseFileOption].FirstOrDefault()?.Text;
 
             HttpResponseMessage response = await programState.Client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
-            await HandleResponseAsync(programState, commandInput, shellState.ConsoleManager, response, programState.EchoRequest, headersTarget, bodyTarget, cancellationToken).ConfigureAwait(false);
+            await HandleResponseAsync(programState, commandInput, shellState.ConsoleManager, response, programState.EchoRequest, headersTarget, bodyTarget, _fileSystem, cancellationToken).ConfigureAwait(false);
         }
 
-        private static async Task HandleResponseAsync(HttpState programState, DefaultCommandInput<ICoreParseResult> commandInput, IConsoleManager consoleManager, HttpResponseMessage response, bool echoRequest, string headersTargetFile, string bodyTargetFile, CancellationToken cancellationToken)
+        private static async Task HandleResponseAsync(HttpState programState, DefaultCommandInput<ICoreParseResult> commandInput, IConsoleManager consoleManager, HttpResponseMessage response, bool echoRequest, string headersTargetFile, string bodyTargetFile, IFileSystem fileSystem, CancellationToken cancellationToken)
```

Comment:
Does this need to be passed in? Can we not use _fileSystem directly?

Reply:
It needed to be passed because the method was static. But it didn't need to be static, so I changed it to be an instance method and took out the parameter.